storage/grpc_metrics.go

```diff
@@ -0,0 +1,295 @@
+	endpointToUse := determineMonitoringEndpoint(config.endpoint)
+	if endpointToUse != "" {
+		meOpts = append(meOpts, mexporter.WithMonitoringClientOptions(option.WithEndpoint(endpointToUse)))
+	}
+	exporter, err := mexporter.New(meOpts...)
+	if err != nil {
+		return nil, err
+	}
+	metricViews := []metric.View{
+		createHistogramView("grpc.client.attempt.duration", "A view of grpc.client.attempt.duration with histogram boundaries more appropriate for Google Cloud Storage RPCs", "s", latencyHistogramBoundaries()),
+		createHistogramView("grpc.client.attempt.rcvd_total_compressed_message_size", "A view of grpc.client.attempt.rcvd_total_compressed_message_size with histogram boundaries more appropriate for Google Cloud Storage RPCs", "By", sizeHistogramBoundaries()),
+		createHistogramView("grpc.client.attempt.sent_total_compressed_message_size", "A view of grpc.client.attempt.sent_total_compressed_message_size with histogram boundaries more appropriate for Google Cloud Storage RPCs", "By", sizeHistogramBoundaries()),
+	}
+	provider := metric.NewMeterProvider(
+		metric.WithReader(metric.NewPeriodicReader(&exporterLogSuppressor{exporter: exporter}, metric.WithInterval(time.Minute))),
+		metric.WithResource(preparedResource.resource),
+		metric.WithView(metricViews...),
+	)
+	mo := opentelemetry.MetricsOptions{
+		MeterProvider: provider,
+		Metrics: opentelemetry.DefaultMetrics().Add(
+			"grpc.lb.wrr.rr_fallback",
+			"grpc.lb.wrr.endpoint_weight_not_yet_usable",
+			"grpc.lb.wrr.endpoint_weight_stale",
+			"grpc.lb.wrr.endpoint_weights",
+			"grpc.lb.rls.cache_entries",
+			"grpc.lb.rls.cache_size",
+			"grpc.lb.rls.default_target_picks",
+			"grpc.lb.rls.target_picks",
+			"grpc.lb.rls.failed_picks"),
+		OptionalLabels: []string{"grpc.lb.locality"},
+	}
+	do := option.WithGRPCDialOption(opentelemetry.DialOption(opentelemetry.Options{MetricsOptions: mo}))
+	context := &internalMetricsContext{
+		project:    preparedResource.projectToUse,
+		endpoint:   endpointToUse,
+		clientOpts: []option.ClientOption{do},
+		provider:   provider,
+		close:      createShutdown(ctx, provider),
+	}
+	return context, nil
+}
+
+func enableClientMetrics(ctx context.Context, s *settings) (*internalMetricsContext, error) {
+	_, ep, err := htransport.NewClient(ctx, s.clientOption...)
+	if err != nil {
+		return nil, fmt.Errorf("gRPC Metrics: %w", err)
+	}
+	project := ""
+	c, err := transport.Creds(ctx, s.clientOption...)
+	if err == nil {
+		project = c.ProjectID
+	}
+	// Enable client-side metrics for gRPC
+	metricsContext, err := newGRPCMetricContext(ctx, internalMetricsConfig{
+		project:  project,
+		endpoint: ep,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("gRPC Metrics: %w", err)
+	}
+	return metricsContext, nil
+}
+
+func createShutdown(ctx context.Context, provider *metric.MeterProvider) func() {
+	return func() {
+		provider.Shutdown(ctx)
+	}
+}
+
+// Silences permission errors after initial error is emitted to prevent
```

Comment:
Maybe explain that this implements the interface for `Exporter` as well? Otherwise it's confusing what's going on here.